specification/resources/resource-manager/Microsoft.Resources/stable/2024-04-01/resources.json

```diff
@@ -5440,16 +5488,26 @@
       },
       "description": "Deployment on error behavior with additional details."
     },
-    "DeploymentValidateResult": {
+    "DeploymentValidateError": {
       "properties": {
         "error": {
-          "readOnly": true,
-          "$ref": "../../../../../common-types/resource-management/v1/types.json#/definitions/ErrorResponse",
-          "description": "The deployment validation error."
-        },
+          "$ref": "../../../../../common-types/resource-management/v2/types.json#/definitions/ErrorResponse"
```

Comment:
Out of interest, why remove the `readOnly` annotation?

Reply:
it had x-ms-external before, can add `readOnly` though, the only challenge is that there's 99% of the definitions without it, so will add to on this one only.